crates/hyperqueue/src/server/autoalloc/process.rs

```diff
@@ -1,255 +1,646 @@
-            _ => return,
+async fn handle_message(
+    autoalloc: &mut AutoAllocState,
+    state_ref: &StateRef,
+    message: AutoAllocMessage,
+) -> ControlFlow<()> {
+    let queue_id: Option<QueueId> = {
+        match message {
+            AutoAllocMessage::WorkerConnected(id, manager_info) => {
+                log::debug!(
+                    "Registering worker {id} for allocation {}",
+                    manager_info.allocation_id
+                );
+                on_worker_connected(state_ref, autoalloc, id, &manager_info);
+                autoalloc.get_queue_id_by_allocation(&manager_info.allocation_id)
+            }
+            AutoAllocMessage::WorkerLost(id, manager_info, reason) => {
+                log::debug!(
+                    "Removing worker {id} from allocation {}",
+                    manager_info.allocation_id
+                );
+                on_worker_lost(state_ref, autoalloc, id, &manager_info, reason);
+                autoalloc.get_queue_id_by_allocation(&manager_info.allocation_id)
+            }
+            AutoAllocMessage::JobCreated(_) => None,
+            AutoAllocMessage::GetQueues(response) => {
+                let queues: Map<QueueId, QueueData> = autoalloc
+                    .queues()
+                    .map(|(id, queue)| {
+                        (
+                            id,
+                            QueueData {
+                                info: queue.info().clone(),
+                                name: queue.name().map(|name| name.to_string()),
+                                manager_type: queue.manager().clone(),
+                            },
+                        )
+                    })
+                    .collect();
+                response.respond(queues);
+                None
+            }
+            AutoAllocMessage::AddQueue {
+                server_directory,
+                manager,
+                params,
+                response,
+            } => {
+                let result = create_queue(autoalloc, state_ref, server_directory, manager, params);
+                let queue_id = result.as_ref().ok().copied();
+                response.respond(result);
+                queue_id
+            }
+            AutoAllocMessage::RemoveQueue {
+                id,
+                force,
+                response,
+            } => {
+                let result = remove_queue(autoalloc, state_ref, id, force).await;
+                response.respond(result);
+                None
+            }
+            AutoAllocMessage::GetAllocations(queue_id, response) => {
+                let result = match autoalloc.get_queue(queue_id) {
+                    Some(queue) => Ok(queue.all_allocations().cloned().collect()),
+                    None => Err(anyhow::anyhow!("Queue {queue_id} not found")),
+                };
+                response.respond(result);
+                None
+            }
+            AutoAllocMessage::Stop(response) => {
```

Comment:
Do we need explicit stop? Maybe just throw away the sending part?

Reply:
Good point.